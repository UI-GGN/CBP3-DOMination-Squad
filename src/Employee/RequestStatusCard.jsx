import { Card, CardContent, CardActions, Button } from '@mui/material';

const RequestStatusCard = ({ status, color, date }) => {
  return (
    <Card className="m-5" sx={{ width: 250, display: 'flex', flexDirection: 'column' }}>
      <CardContent>
        <div className="flex flex-row justify-between">
          <div className="text-sm text-gray-500">Status</div>
          <div className="text-sm text-gray-500">{date}</div>
        </div>
        <p className="mb-2">{status}</p>
        <div className={`w-full h-1 ${color}`}></div>
      </CardContent>
      <CardActions>
        <Button size="small">More details</Button>
      </CardActions>
    </Card>
  );
};

export default RequestStatusCard;
