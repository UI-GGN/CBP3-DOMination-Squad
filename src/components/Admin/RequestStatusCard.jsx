import { Card, CardContent, CardActions, Button } from '@mui/material';

const RequestStatusCard = ({ name, date, pickup, time }) => {
  return (
    <Card className="m-3 basis-1/6 shrink-0" sx={{ width: 275, display: 'flex', flexDirection: 'column' }}>
      <CardContent>
        <div>
          <div className="flex flex-row justify-between items-center">
            <div className="text-sm text-gray-500">Name</div>
            <div className="text-sm text-gray-500">Date</div>
          </div>
          <div className="flex flex-row justify-between items-center mb-5">
            <div className="text-base font-bold">{name}</div>
            <div className="text-base ">{date}</div>
          </div>
          <div className="flex flex-row justify-between">
            <div className="text-sm text-gray-500">Pickup </div>
            <div className="text-sm text-gray-500">Time</div>
          </div>
          <div className="flex flex-row justify-between">
            <div className="text-base w-2/3">{pickup}</div>
            <div className="text-base ">{time}</div>
          </div>
        </div>
      </CardContent>
      <CardActions disableSpacing sx={{ mt: 'auto' }}>
        <Button size="small">Assign Route</Button>
      </CardActions>
    </Card>
  );
};

export default RequestStatusCard;
