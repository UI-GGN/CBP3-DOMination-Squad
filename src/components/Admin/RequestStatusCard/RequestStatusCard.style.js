import Card from '../../common/Card';
import styled from 'styled-components';

export const CardDetailsContainer = styled(Card)`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export const HeaderContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

export const DetailsContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

export const HeaderText = styled.h1`
  font-size: 14px;
  line-height: 20px;
  color: rgb(156, 163, 175);
`;

export const Button = styled.button`
  background-color: white;
  font-weight: 600;
  font-size: 16px;
  color: #17252a;
  margin-top: 4px;
  color: ${(props) => (props?.color ? props.color : '#17252a')};
  &:hover {
    font-weight: 700;
  }
`;

export const HorizontalLine = styled.div`
  height: 0.5px;
  background-color: rgb(229, 231, 235);
  margin-top: 12px;
  margin-bottom: 12px;
`;
