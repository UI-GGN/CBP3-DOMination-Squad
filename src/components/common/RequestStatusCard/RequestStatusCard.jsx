import circle from '../../../assets/circle_icon.png';
import destination from '../../../assets/destination_icon.png';
import {
  CardDetailsContainer,
  DetailsContainer,
  LocationSegment,
  LocationPathIconSegment,
  LocationPathIcon,
  LocationDetails,
  HeaderContainer,
  HeaderText,
  DetailsText,
  HighlightedText,
  Button,
  HorizontalLine,
  VerticalLine,
} from './RequestStatusCard.style.js';

const RequestStatusCard = ({ name, employeeID, date, time, pickup, drop }) => {
  return (
    <CardDetailsContainer>
      <HeaderContainer>
        <HeaderText>Name</HeaderText>
      </HeaderContainer>
      <DetailsContainer>
        <DetailsText>
          {name} ({employeeID})
        </DetailsText>
      </DetailsContainer>
      <HorizontalLine />

      <HeaderContainer>
        <HeaderText>Date </HeaderText>
        <HeaderText>Time</HeaderText>
      </HeaderContainer>
      <DetailsContainer>
        <DetailsText>{date}</DetailsText>
        <HighlightedText>{time}</HighlightedText>
      </DetailsContainer>
      <HorizontalLine />

      <LocationSegment>
        <LocationPathIconSegment>
          <LocationPathIcon>
            <img src={circle} style={{ maxHeight: '70%' }} />
          </LocationPathIcon>
          <VerticalLine />
          <LocationPathIcon>
            <img src={destination} style={{ maxHeight: '70%' }} />
          </LocationPathIcon>
        </LocationPathIconSegment>
        <LocationDetails>
          <HeaderText>Start Point</HeaderText>
          <DetailsText style={{ marginBottom: 10 }}>{pickup}</DetailsText>
          <HeaderText style={{ marginTop: 10 }}>End Point</HeaderText>
          <DetailsText>{drop}</DetailsText>
        </LocationDetails>
      </LocationSegment>
      <HorizontalLine />

      <HeaderContainer>
        <Button color="#3aafa9">Assign Route</Button>
        <Button color="#d22b2b" marginTop="4px">
          Reject Request
        </Button>
      </HeaderContainer>
    </CardDetailsContainer>
  );
};

export default RequestStatusCard;
