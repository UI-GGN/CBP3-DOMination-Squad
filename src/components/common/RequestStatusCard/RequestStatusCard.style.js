import Card from '../Card/index.js';
import styled from 'styled-components';

export const CardDetailsContainer = styled(Card)`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  flex-grow: 1;

  @media (max-width: 768px) {
    flex-grow: 1;
  }
`;

export const Header = styled.div`
  display: flex;
  flex-direction: column;
`;

export const DateSection = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Location = styled.div`
  display: flex;
  flex-direction: column;
`;

export const TitleContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

export const DetailsContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

export const LocationSegment = styled.div`
  display: flex;
  flex-direction: row;
  flex: 1;
  margin: 1px;
`;

export const LocationPathIconSegment = styled.div`
  display: flex;
  flex-direction: column;
  width: 10%;
  align-items: center;
`;

export const LocationPathIcon = styled.div`
  display: flex;
  height: 20%;
  align-items: center;
`;

export const LocationPathImage = styled.img`
  max-height: 70%;
  max-width: 100%;
  align-self: center;

  @media (max-width: 800px) {
    max-width: 80%;
  }
`;

export const LocationDetails = styled.div`
  display: flex;
  flex-direction: column;
  width: 90%;
  justify-content: space-between;
`;

export const HighlightedText = styled.h1`
  font-weight: 700;
  line-height: 24px;
`;

export const TitleText = styled.h1`
  font-size: 14px;
  line-height: 20px;
  color: rgb(156, 163, 175);
`;

export const DetailsText = styled.h1`
  font-size: 16px;
  line-height: 24px;
`;

export const HorizontalLine = styled.div`
  height: 0.5px;
  background-color: rgb(229, 231, 235);
  margin-top: 10px;
  margin-bottom: 10px;
`;

export const VerticalLine = styled.div`
  height: 60%;
  border-style: dotted;
  border-left-width: 5px;
`;
