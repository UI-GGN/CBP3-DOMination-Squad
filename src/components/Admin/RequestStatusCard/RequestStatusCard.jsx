import { CardDetailsContainer, DetailsContainer, HeaderContainer, HeaderText, Button } from './RequestStatusCard.style';

const RequestStatusCard = ({ user, name, employeeID, date, time, pickup, drop }) => {
  if (user.type === 'admin') {
    return (
      <CardDetailsContainer>
        <HeaderContainer>
          <HeaderText>Name</HeaderText>
          <HeaderText>Employee ID</HeaderText>
        </HeaderContainer>
        <DetailsContainer>
          <div className="text-base font-bold">{name}</div>
          <div className="text-base ">{employeeID}</div>
        </DetailsContainer>
        <HeaderContainer>
          <HeaderText>Date </HeaderText>
          <HeaderText>Time</HeaderText>
        </HeaderContainer>
        <DetailsContainer>
          <div className="text-base w-2/3">{date}</div>
          <div className="text-base ">{time}</div>
        </DetailsContainer>
        <HeaderContainer>
          <HeaderText>Pickup</HeaderText>
          <HeaderText>Drop</HeaderText>
        </HeaderContainer>
        <DetailsContainer>
          <div className="text-base font-bold text-left">{pickup}</div>
          <div className="text-base text-right">{drop}</div>
        </DetailsContainer>

        <HeaderContainer>
          <Button color="#3aafa9">Assign Route</Button>
          <Button color="#d22b2b" marginTop="4px">
            Reject Request
          </Button>
        </HeaderContainer>
      </CardDetailsContainer>
    );
  } else {
    return (
      <CardDetailsContainer>
        <HeaderContainer>
          <HeaderText>Name</HeaderText>
          <HeaderText>Date</HeaderText>
        </HeaderContainer>
        <div className="flex flex-row justify-between mb-5">
          <div className="text-base font-bold">{name}</div>
          <div className="text-base ">{date}</div>
        </div>
        <HeaderContainer>
          <HeaderText>Pickup </HeaderText>
          <HeaderText>Time</HeaderText>
        </HeaderContainer>
        <HeaderContainer>
          <div className="text-base w-2/3">{pickup}</div>
          <div className="text-base ">{time}</div>
        </HeaderContainer>

        <HeaderContainer>
          <Button color="#d22b2b" marginTop="4px">
            Cancel Request
          </Button>
        </HeaderContainer>
      </CardDetailsContainer>
    );
  }
};

export default RequestStatusCard;
