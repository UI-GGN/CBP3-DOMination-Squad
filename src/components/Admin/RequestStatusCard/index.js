import RequestStatusCard from './RequestStatusCard';

export default RequestStatusCard;
