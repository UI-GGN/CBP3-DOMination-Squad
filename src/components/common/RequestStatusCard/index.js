import RequestStatusCard from './RequestStatusCard.jsx';

export default RequestStatusCard;
