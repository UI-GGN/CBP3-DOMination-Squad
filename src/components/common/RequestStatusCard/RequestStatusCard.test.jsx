import RequestStatusCard from './RequestStatusCard';
import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

const props = {
  name: 'Shivansh Garg',
  employeeID: '33067',
  projectCode: '123',
  date: '2023-07-03',
  time: '10:00 AM',
  location: true,
  pickup: 'Home',
  drop: 'Office',
};

describe('Card', () => {
  it('should render RequestStatusCard component', () => {
    const { container, getByText, getByTestId } = render(<RequestStatusCard {...props} />);

    expect(getByTestId('employee-details-section')).toBeDefined();
    expect(getByTestId('location-section')).toBeDefined();

    expect(getByText('Shivansh Garg (33067)')).toBeInTheDocument();
    expect(getByTestId('name-label').textContent).toBe('Name');
    expect(getByTestId('name').textContent).toBe('Shivansh Garg (33067)');

    expect(getByTestId('project-code-label').textContent).toBe('Project Code');
    expect(getByTestId('project-code').textContent).toBe('123');

    expect(getByTestId('date-label').textContent).toBe('Date');
    expect(getByTestId('date').textContent).toBe('2023-07-03');

    expect(getByTestId('pickup-time-label').textContent).toBe('Pickup Time');
    expect(getByTestId('pickup-time').textContent).toBe('10:00 AM');

    expect(getByText('Start Point')).toBeInTheDocument();
    expect(getByTestId('start-point-label')).toHaveTextContent('Start Point');
    expect(getByTestId('pickup-location')).toHaveTextContent('Home');

    expect(getByText('End Point')).toBeInTheDocument();
    expect(getByTestId('end-point-label')).toHaveTextContent('End Point');
    expect(getByTestId('drop-location')).toHaveTextContent('Office');
    expect(container).toMatchSnapshot();
  });

  it('should render RequestStatusCard with custom footer component', () => {
    const updatedProps = {
      ...props,
      footer: () => <div>Custom Footer</div>,
    };

    const { getByText } = render(<RequestStatusCard {...updatedProps} />);

    expect(getByText('Custom Footer')).toBeInTheDocument();
  });

  it('should not render employee detials section with missing name prop', () => {
    const updatedProps = {
      ...props,
      name: null,
    };

    const { queryByTestId } = render(<RequestStatusCard {...updatedProps} />);

    expect(queryByTestId('employee-details-section')).toBeNull();
    expect(queryByTestId('name-label')).toBeNull();
    expect(queryByTestId('name')).toBeNull();
    expect(queryByTestId('project-code-label')).toBeNull();
    expect(queryByTestId('project-code')).toBeNull();
  });

  it('should not render location segment when location prop is false', () => {
    const updatedProps = {
      ...props,
      location: false,
    };

    const { queryByTestId } = render(<RequestStatusCard {...updatedProps} />);

    expect(queryByTestId('location-section')).toBeNull();
    expect(queryByTestId('start-point-label')).toBeNull();
    expect(queryByTestId('pickup-location')).toBeNull();
    expect(queryByTestId('end-point-label')).toBeNull();
    expect(queryByTestId('drop-location')).toBeNull();
  });
});
