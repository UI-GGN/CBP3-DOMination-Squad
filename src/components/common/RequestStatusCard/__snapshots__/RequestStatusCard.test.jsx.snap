// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > should render RequestStatusCard component 1`] = `
<div>
  <div
    class="sc-gueYoa cNueUJ sc-dmqHEX cbUebc"
  >
    <div
      class="sc-hLseeU jMASXh"
      data-testid="employee-details-section"
    >
      <div
        class="sc-gLDzan hmbCMJ"
      >
        <h1
          class="sc-fbJfA cedOEj"
          data-testid="name-label"
        >
          Name
        </h1>
        <h1
          class="sc-fbJfA cedOEj"
          data-testid="project-code-label"
        >
          Project Code
        </h1>
      </div>
      <div
        class="sc-iAEyYk czncTD"
      >
        <h1
          class="sc-brKeYL dwsKPC"
          data-testid="name"
        >
          Shivansh Garg
           (
          33067
          )
        </h1>
        <h1
          class="sc-brKeYL dwsKPC"
          data-testid="project-code"
        >
          123
        </h1>
      </div>
      <div
        class="sc-ktEKTO kAVaEc"
      />
    </div>
    <div
      class="sc-eDDNvR duEbAE"
    >
      <div
        class="sc-gLDzan hmbCMJ"
      >
        <h1
          class="sc-fbJfA cedOEj"
          data-testid="date-label"
        >
          Date
        </h1>
        <h1
          class="sc-fbJfA cedOEj"
          data-testid="pickup-time-label"
        >
          Pickup Time
        </h1>
      </div>
      <div
        class="sc-iAEyYk czncTD"
      >
        <h1
          class="sc-brKeYL dwsKPC"
          data-testid="date"
        >
          2023-07-03
        </h1>
        <h1
          class="sc-csCMJt cfDkEU"
          data-testid="pickup-time"
        >
          10:00 AM
        </h1>
      </div>
      <div
        class="sc-ktEKTO kAVaEc"
      />
    </div>
    <div
      class="sc-jTrPJq cvChAg"
      data-testid="location-section"
    >
      <div
        class="sc-fsQiph fzWrD"
      >
        <div
          class="sc-qRumB hkIvZd"
        >
          <div
            class="sc-jsMahE iQgzMo"
          >
            <img
              class="sc-kFuwaP llEZif"
              src="/src/assets/circle_icon.png"
            />
          </div>
          <div
            class="sc-hAtEyd bEtTip"
          />
          <div
            class="sc-jsMahE iQgzMo"
          >
            <img
              class="sc-kFuwaP llEZif"
              src="/src/assets/destination_icon.png"
            />
          </div>
        </div>
        <div
          class="sc-irTswW gTXNtA"
        >
          <h1
            class="sc-fbJfA cedOEj"
            data-testid="start-point-label"
          >
            Start Point
          </h1>
          <h1
            class="sc-brKeYL dwsKPC"
            data-testid="pickup-location"
            style="margin-bottom: 10px;"
          >
            Home
          </h1>
          <h1
            class="sc-fbJfA cedOEj"
            data-testid="end-point-label"
            style="margin-top: 10px;"
          >
            End Point
          </h1>
          <h1
            class="sc-brKeYL dwsKPC"
            data-testid="drop-location"
          >
            Office
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
